### ISSUE TITLE
IO failed for 100GB file upload

### ISSUE THREAD

**Issue by akmithal** [noobaa/noobaa-core#6612]
### Environment info
- NooBaa Version: master-20210627
- Platform: OCP 4.6.16

### Actual behavior
1. IO upload failed for 100GB file

### Expected behavior
1. IO upload should not fail

### Steps to reproduce
1. Installed new code - master-20210627
2. Started upload of 100 GB file

Got following error:

`upload failed: ./big_file_100GB to s3://bucket-1/big_file_100GB Connection was closed before we received a valid response from endpoint URL: "https://s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com/bucket-1/big_file_100GB?uploadId=ad036aae-cff1-46d8-9775-48ba0385c939&partNumber=355".`

```
[Endpoint/8]    [L0] core.sdk.namespace_fs:: NamespaceFS: abort_object_upload /nsfs/nsfs-nsr-1/b
ucket-1/.noobaa-nsfs_60d966a816cfee0022497622/multipart-uploads/ad036aae-cff1-46d8-9775-48ba0385c939
2021-06-28 08:44:25.317614 [PID-8/TID-8] [L1] FS::FSWorker::Begin: Readdir _path=/nsfs/nsfs-nsr-1/bucket-1/.noobaa-nsfs_60d966a816cfee0022497622/multipart-uploads/ad036aae-cff1-46d8-9775-48ba0385c
939 
2021-06-28 08:44:25.317653 [PID-8/TID-8] [L1] FS::Stat::OnOK: _path=/nsfs/nsfs-nsr-1/bucket-1/.noobaa-nsfs_60d966a816cfee0022497622/multipart-uploads/ad036aae-cff1-46d8-9775-48ba0385c939/part-444 
_stat_res.st_ino=101448123 _stat_res.st_size=237568 
2021-06-28 08:44:25.317669 [PID-8/TID-17] [L1] FS::FSWorker::Execute: Readdir _path=/nsfs/nsfs-nsr-1/bucket-1/.noobaa-nsfs_60d966a816cfee0022497622/multipart-uploads/ad036aae-cff1-46d8-9775-48ba03
85c939 _uid=5000 _gid=800 _backend=GPFS 
```

**Comment by akmithal** [comment]
[noobaa_diagnostics_1624870664.tar.gz](https://github.com/noobaa/noobaa-core/files/6724907/noobaa_diagnostics_1624870664.tar.gz)

**Comment by akmithal** [comment]
It is easily recreatable. Defect 6602 was attempted with KEEP_ALIVE value of 10. On 27th June build, it is 15. 

`config.S3_KEEP_ALIVE_WHITESPACE_INTERVAL = 15 * 1000;`

**Comment by jeniawhite** [comment]
@akmithal 
Seems like the version is valid and the keep alive should be there but I do not see it.
I can see that you have collisions in the database for the buckets which clutter the logging, would be great if you could attach a PostgreSQL DB dump as well.

**Comment by nimrod-becker** [comment]
Collisions have been fixed with
#6657 and #6654

**Comment by akmithal** [comment]
Closing this bug as this seems to be fixed in master-20210802 build.
